### PR TITLE
Resolving SonarCloud code smells - round 1

### PR DESCRIPTION
**Pull request by samsmithnz**
Primary goals are to
- simply boolean comparisons. e.g. [expression] == true will become [expression], and [expression] == false will become ![expression]
- fixed short circuits to always be && or || instead of & or |
- Added stringbuilder in a few places to replace string concat - should improve performance.

### CODE REVIEW

File: src/AzurePipelinesToGitHubActionsConverter.Core/Conversion.cs [hunk 1]
```diff
@@ -185,30 +181,30 @@ private ConversionResponse ConvertAzurePipelineToGitHubActionV2(string yaml, boo
 
                 //Strategy
                 JsonElement strategy = new JsonElement();
-                if (jsonObject.TryGetProperty("strategy", out jsonElement) == true)
+                if (jsonObject.TryGetProperty("strategy", out jsonElement) )
```
Comment (f2calv) [inline src/AzurePipelinesToGitHubActionsConverter.Core/Conversion.cs:184]: would be nice to autoformat the changed files and remove the extra whitespace between final brackets

File: src/AzurePipelinesToGitHubActionsConverter.Core/Conversion.cs [hunk 2]
```diff
@@ -185,30 +181,30 @@ private ConversionResponse ConvertAzurePipelineToGitHubActionV2(string yaml, boo
...
                 //If we have stages, convert them into jobs first:
-                if (jsonObject.TryGetProperty("stages", out jsonElement) == true)
+                if (jsonObject.TryGetProperty("stages", out jsonElement) )
                 {
                     StagesProcessing sp = new StagesProcessing(_verbose);
                     gitHubActions.jobs = sp.ProcessStagesV2(jsonElement, strategy);
                 }
                 //If we don't have stages, but have jobs:
-                else if (jsonObject.TryGetProperty("stages", out jsonElement) == false & jsonObject.TryGetProperty("jobs", out jsonElement) == true)
+                else if (jsonObject.TryGetProperty("stages", out jsonElement) == false && jsonObject.TryGetProperty("jobs", out jsonElement) )
```
Comment (f2calv) [inline src/AzurePipelinesToGitHubActionsConverter.Core/Conversion.cs:196]: why not refactor jsonObject.TryGetProperty("stages", out jsonElement) == false to !(jsonObject.TryGetProperty("stages", out jsonElement)) ?

Comment (samsmithnz) [reply src/AzurePipelinesToGitHubActionsConverter.Core/Conversion.cs]: that was next! Just did exactly that.
:)

File: src/AzurePipelinesToGitHubActionsConverter.Core/Conversion.cs (hunk 1 above)
Comment (samsmithnz) [reply src/AzurePipelinesToGitHubActionsConverter.Core/Conversion.cs]: I usually do - missed this one. Thank you.

**Pull request merged**